src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -403,9 +409,106 @@ void Syscollector::start()
         m_spSyncProtocol->reset();
     }
 
+    lock.unlock(); // Unlock to allow stop during notifyDataClean
```

Comment:
I think we can improve this mutex usage.

Reply:
Done